doc/source/whatsnew/v1.5.0.rst

```diff
@@ -57,6 +57,39 @@ Styler
 
 - Fixed bug in :class:`CSSToExcelConverter` leading to ``TypeError`` when border color provided without border style for ``xlsxwriter`` engine (:issue:`42276`)
 
+.. _whatsnew_150.notable_bug_fixes.groupby_transform_dropna:
+
+Using ``dropna=True`` with ``groupby`` transformers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A transformer is an operation whose result has the same size as its input. When the
+result is a :class:`DataFrame` or :class:`Series`, it is also required that the
+index of the result is equal to that of the input. In pandas 1.4, using
```

Comment:
"is equal" -> "matches"?

I think there was a recent issue with something like df.groupby(foo).transform(DataFrame.sort_index) where there was some question as to whether the index should be retained.  Did that get sorted out? (Or is it not relevant here?)

Reply:
It didn't yet get sorted out. In fact, I went to improve the docs because of that issue, and found that `dropna` was not well supported for transforms. Wanted to first improve this before returning to the docs.

I agree matches is better than equal here - and plan to clarify that `groupby(...).transform` will align when possible.